packages/story-editor/src/components/floatingMenu/elements/borderWidthAndColor.js

```
@@ -19,19 +19,110 @@
 
 /**
  * Internal dependencies
  */
-import { IconButton } from './shared';
+import { useStory } from '../../../app';
+import { canSupportMultiBorder } from '../../../masks';
+import { DEFAULT_BORDER } from '../../panels/design/border/shared';
+import { Input, Color, useProperties } from './shared';
+
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const Dash = styled.div`
+  width: 8px;
+  height: 1px;
+  background-color: ${({ theme }) => theme.colors.border.defaultNormal};
+`;
+
+function getHasBorderWidth(border) {
+  if (!border) {
+    return false;
+  }
+  const { left, right, top, bottom } = border;
+  return Boolean(left || right || top || bottom);
+}
+
+const BLACK = { color: { r: 0, g: 0, b: 0 } };
 
 function BorderWidthAndColor() {
+  // Note that "mask" never updates on an element,
+  // so selecting it cannot cause re-renders
+  // We need it to determine if border opacity is allowed.
+  const { border = DEFAULT_BORDER, mask } = useProperties(['border', 'mask']);
+  const updateSelectedElements = useStory(
+    (state) => state.actions.updateSelectedElements
+  );
+
+  // Only multi-border elements support border opacity
+  const canHaveBorderOpacity = canSupportMultiBorder({ mask });
+
+  // We only allow editing the current border width, if all borders are identical
+  const hasUniformBorder =
+    border.left === border.right &&
+    border.left === border.top &&
+    border.left === border.bottom;
+
+  // We only allow editing border color, if at least one border has a non-zero width
+  const hasBorderWidth = getHasBorderWidth(border);
+
+  // If both controls are displayed, also add a dash between them
+  const hasBoth = hasUniformBorder && hasBorderWidth;
+
+  // NB: Note that it can never be the case, that neither input is shown, so at least
+  // one of them will always render.
+
+  const handleWidthChange = (value) =>
+    updateSelectedElements({
+      properties: ({ border: oldBorder }) => ({
```

Hm, the default black border in the Design panel comes from the `DEFAULT_BORDER`, so it looks like `oldBorder` doesn't include the color here for some reason? When getting the initial border value, in theory, it should be black by default when using `DEFAULT_BORDER`.

Maybe `const { border = DEFAULT_BORDER, mask } = useProperties(['border', 'mask']);` above is not getting `border` value back as `undefined` / `null` and thus the `DEFAULT_BORDER` is not assigned correctly?

---

The `border` variable is not used inside the `handleWidthChange` logic. But I can explicitly set it - done in 8cf3b8eb92.